packages/arb-bridge-eth/contracts/arch/Hashing.sol

```diff
@@ -42,35 +42,49 @@ library Hashing {
         return res;
     }
 
+    /*
+     * Note that dataLength has to be a power of two.
```

Comment:
Can we add a require/revert for this? Seems like otherwise this is pretty risky. Alternatively, this should be private to make it clear that the API is unsafe. One third option would be to pass in `n` instead of `2^n`